app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/db/DatabaseHelper.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package android.support.v17.leanback.supportleanbackshowcase.app.room.db;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.persistence.room.Room;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.support.annotation.Nullable;
+import android.support.annotation.WorkerThread;
+import android.support.v17.leanback.supportleanbackshowcase.app.room.db.entity.VideoEntity;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static android.support.v17.leanback.supportleanbackshowcase.app.room.db.AppDatabase.DATABASE_NAME;
+
+/**
+ * Helper class to create/ update the database
+ */
+public class DatabaseHelper {
+
+    private static final String RENTED = "rented";
+    private static final String STATUS = "status";
+    private static final String CARD = "card";
+    private static final String BACKGROUND = "background";
+    private static final String VIDEO = "video";
+
+    private static DatabaseHelper sInstance;
+
+    private final MutableLiveData<Boolean> mIsDatabaseCreated = new MutableLiveData<>();
+
+    private final MutableLiveData<Boolean> mDatabaseUpdatedSignal = new MutableLiveData<>();
+
+    private AppDatabase mDb;
+
+    /**
+     * use atomic boolean for synchronization
+     */
+    private final AtomicBoolean mInitialized = new AtomicBoolean(false);
+
+
+    public synchronized static DatabaseHelper getInstance() {
+        if (sInstance == null) {
+            sInstance = new DatabaseHelper();
+        }
+        return sInstance;
+    }
+
+    public LiveData<Boolean> isDatabaseCreated() {
```

Comment:
you already keep everything in db, why do you need a live data for initialization?

Reply:
Hi, Yigit. Correct me if I'm wrong. But let's imagine a scenario that the database is huge and takes a relatively long time to create. Under this circumstance, live data cannot go to data base directly to fetch the result (this is when the "ABSENT" the null live data should be returned), so I use databaseCreated as a signal to notify wen the database is ready to access.

Reply:
What do you mean it is huge? Creating the DB is trivial. If you are talking about the case where items are not fetched yet, then it is OK for LiveData, it will just return empty set (or null if querying 1 item).

Reply:
yes, you are absolutely right, it is a redundant design. Removed